pkg/util/docker/network.go

```diff
@@ -165,3 +173,55 @@ func DefaultGateway() (net.IP, error) {
 	}
 	return ip, nil
 }
+
+// GetAgentContainerNetworkMode provides the network mode of the Agent container
+func GetAgentContainerNetworkMode() (string, error) {
```

Comment:
nit: could we add a comment to say that if you need this info, consider using `util.GetAgentNetworkMode` instead to benefit from the cache? Otherwise we might use this method in a non-optimal way.

Reply:
makes sense, done 👍